resources/sip11/ra/src/test/java/org/mobicents/slee/resource/sip11/test/SipResourceAdaptorTest.java

```diff
@@ -0,0 +1,144 @@
+		Answer<Object> answer = new Answer<Object>() {
+			public Object answer(InvocationOnMock invocation) {
+				System.out.println(" (TRACER-MOCK) :: info() called :: " + Arrays.toString(invocation.getArguments()));
+				return null;
+			}
+		};
+		doAnswer(answer).when(tracer1).info(anyString());
+	}
+
+	// minimal mock just to satisfy SipResourceAdaptor's usage of the tracer
+	private static void mockRaContext() {
+		when(raContext1.getEntityName()).thenReturn("stack#1");
+		when(raContext2.getEntityName()).thenReturn("stack#2");
+	}
+
+	@BeforeClass
+	public static void prepareEnvSharedByAllTests() {
+		mockTracer();
+		mockRaContext();
+	}
+
+	@Test
+	/*
+	 * Purpose of this test it to verify whether two RA entities can be activated on the
+	 * same network interface within same class loader.
+	 */
+	public void testActivateTwoRaEntities() {
+		// create two RAs
+		SipResourceAdaptor ra1 = new SipResourceAdaptor();
+		SipResourceAdaptor ra2 = new SipResourceAdaptor();
+
+		// initialize internal dependencies of RA#1
+		Whitebox.setInternalState(ra1, "tracer", tracer1);
+		Whitebox.setInternalState(ra1, "raContext", raContext1);
+		Whitebox.setInternalState(ra1, "providerWrapper", sleeSipProvider);
+
+		// initialize internal dependencies of RA#2
+		Whitebox.setInternalState(ra2, "tracer", tracer1);
+		Whitebox.setInternalState(ra2, "raContext", raContext2);
+		Whitebox.setInternalState(ra2, "providerWrapper", sleeSipProvider);
+
+		// prepare configuration for RA#1, RA#2
+		ConfigProperties configRa1 = new ConfigProperties();
+		ConfigProperties configRa2;
+
+		configRa1.addProperty(new Property("javax.sip.TRANSPORT", "java.lang.String", "TCP"));
+		configRa1.addProperty(new Property("javax.sip.IP_ADDRESS", "java.lang.String", STACK_ADDRESS));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.BALANCERS", "java.lang.String", ""));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.LoadBalancerHeartBeatingServiceClassName",
+				"java.lang.String", "org.mobicents.ha.javax.sip.LoadBalancerHeartBeatingServiceImpl"));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.LoadBalancerElector", "java.lang.String",
+				"org.mobicents.ha.javax.sip.RoundRobinLoadBalancerElector"));
+		configRa1.addProperty(new Property("org.mobicents.ha.javax.sip.CACHE_CLASS_NAME", "java.lang.String",
+				"org.mobicents.ha.javax.sip.cache.NoCache"));
+
+		// clone common properties
+		configRa2 = (ConfigProperties) configRa1.clone();
+
+		// RA#1 custom configuration
+		configRa1.addProperty(new Property("javax.sip.PORT", "java.lang.Integer", SIP_RA1_PORT));
+		configRa1.addProperty(new Property("javax.sip.STACK_NAME", "java.lang.String", raContext1.getEntityName()));
+
+		// RA#2 custom configuration
+		configRa2.addProperty(new Property("javax.sip.PORT", "java.lang.Integer", SIP_RA2_PORT));
+		configRa2.addProperty(new Property("javax.sip.STACK_NAME", "java.lang.String", raContext2.getEntityName()));
```

Comment:
it seems the raContextX mock are a bit overused as well, can we really just use a literal here, and prevent mocking? We could assert later against the literal (hopefully a constant in the test class)...

I can see raContextX mocking is necessary to complete RA collaborators though...